App/DataType+Utilities.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  DataType+Utilities.swift
+//
+//  Created by Dave Carlson on 2/18/17.
+//  Copyright © 2017 Clinical Cloud Solutions, LLC. All rights reserved.
+//
+
+import Foundation
+import SMART
+
+extension CodeableConcept {
+    
+    func displayString(system: String? = nil) -> String? {
+        // if text is provided, use as default human-readable display
+        if let text = text?.string {
+            return text
+        }
+        
+        var selectedCoding: Coding?
+        
+        if let codeableConcepts = coding {
+            for coding in codeableConcepts {
+                if let codeSystem = system {
+                    // if 'system' parameter included, use first matching Coding
+                    if codeSystem == coding.system?.absoluteString {
+                        selectedCoding = coding
+                        break
+                    }
+                }
+                else {
+                    // use the first coding
+                    selectedCoding = coding
+                    break
+                }
+            }
+        }
+        
+        return selectedCoding?.display?.string ?? selectedCoding?.code?.string
+    }
+    
+}
+
+
+extension Period {
+    
+    func displayString() -> String? {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "MMM dd, yyyy"
```

Comment:
Please just set `dateFormatter.dateStyle` and `dateFormatter.timeStyle`. This way the format will be in the correct locale for the user.